tests/test_ai_synthesis_pass.py

```diff
@@ -0,0 +1,396 @@
+    AIPermutationSynthesis,
+    AILinearFunctionSynthesis,
+    AICliffordSynthesis,
+    AIPauliNetworkSynthesis,
+)
+
+from tests.parametrize_functions import (
+    parametrize_local_mode,
+    parametrize_coupling_map_format,
+)
+
+
+# TODO: For Permutations, the original circuit doesn't return a DAGCircuit with nodes. Decide how the code should behave on this case
+def parametrize_basic_circuit_collector_pass_and_ai_synthesis_pass():
+    return pytest.mark.parametrize(
+        "circuit, collector_pass, ai_synthesis_pass",
+        [
+            # ("basic_swap_circuit", CollectPermutations, AIPermutationSynthesis),
+            ("basic_cnot_circuit", CollectLinearFunctions, AILinearFunctionSynthesis),
+            ("basic_cnot_circuit", CollectCliffords, AICliffordSynthesis),
+            (
+                "random_circuit_transpiled",
+                CollectPauliNetworks,
+                AIPauliNetworkSynthesis,
+            ),
+        ],
+        # ids=["permutation", "linear_function", "clifford"],
+        ids=["linear_function", "clifford", "pauli_network"],
+    )
+
+
+def parametrize_complex_circuit_collector_pass_and_ai_synthesis_pass():
+    return pytest.mark.parametrize(
+        "circuit, collector_pass, ai_synthesis_pass",
+        [
+            (
+                "permutation_circuit_brisbane",
+                CollectPermutations,
+                AIPermutationSynthesis,
+            ),
+            (
+                "linear_function_circuit",
+                CollectLinearFunctions,
+                AILinearFunctionSynthesis,
+            ),
+            ("clifford_circuit", CollectCliffords, AICliffordSynthesis),
+            (
+                "random_pauli_circuit_transpiled",
+                CollectPauliNetworks,
+                AIPauliNetworkSynthesis,
+            ),
+        ],
+        ids=["permutation", "linear_function", "clifford", "pauli_network"],
+    )
+
+
+# TODO: When testing the synthesis with wrong backend, local and cloud behaves differently,
+# so we should decide if this is correct or if we want to unify them
+@parametrize_basic_circuit_collector_pass_and_ai_synthesis_pass()
+def test_ai_local_synthesis_wrong_backend(
+    circuit, collector_pass, ai_synthesis_pass, request
+):
+    if collector_pass == CollectPauliNetworks:
+        pytest.skip("Skipping test for pauli network on local mode")
+
+    original_circuit = request.getfixturevalue(circuit)
+
+    with pytest.raises(
+        PermissionError,
+        match=r"User doesn\'t have access to the specified backend: \w+",
+    ):
+        custom_ai_synthesis_pass = PassManager(
+            [
+                collector_pass(min_block_size=2),
+                ai_synthesis_pass(backend_name="wrong_backend"),
+            ]
+        )
+
+        custom_ai_synthesis_pass.run(original_circuit)
+
+
+# TODO: Tests pass if we add min_block_size=2, max_block_size=27 to collector_pass. If not, tests failed. Confirm why this is happening
```

Comment:
This has to do with the generated circuits we pass to the test. This is not bad per se. The important thing we really need to do i s avoid any randomness in the circuit generation. Either set the seed or load a circuit we design beforehand.

Reply:
Seed fixed: https://github.com/Qiskit/qiskit-ibm-transpiler/pull/128/commits/2195fc0e136116614e04faa5c763cc639635fb71